src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java

```diff
@@ -82,24 +87,69 @@ protected void initBlocking(JsonObject request, JsonObject params) {
     FileDefinition requestFileDefinition = exportRequest.getFileDefinition();
     OkapiConnectionParams okapiConnectionParams = new OkapiConnectionParams(params.mapTo(Map.class));
     FileDefinition fileExportDefinition = createExportFileDefinition(requestFileDefinition);
+    String jobExecutionId = requestFileDefinition.getJobExecutionId();
     String tenantId = okapiConnectionParams.getTenantId();
     Iterator<List<String>> readFileContent = sourceReader.getFileContentIterator(requestFileDefinition, getBatchSize());
     if (readFileContent.hasNext()) {
-      fileDefinitionService.save(fileExportDefinition, tenantId)
-        .compose(savedFileExportDefinition -> {
-          //get executionJob and update exportresult to in-progress
-          String jobExecutionId = requestFileDefinition.getJobExecutionId();
-          initInputDataContext(readFileContent, jobExecutionId);
-          ExportPayload exportPayload = createExportPayload(okapiConnectionParams, savedFileExportDefinition, jobExecutionId);
-          exportNextChunk(exportPayload, readFileContent);
-          return Future.succeededFuture();
-        });
+      fileDefinitionService.save(fileExportDefinition, tenantId).compose(savedFileExportDefinition -> {
+        initInputDataContext(readFileContent, jobExecutionId);
+        ExportPayload exportPayload = createExportPayload(okapiConnectionParams, savedFileExportDefinition, jobExecutionId);
+        exportNextChunk(exportPayload, readFileContent);
+        updateJobExecutionStatusAndExportedFiles(jobExecutionId, fileExportDefinition, tenantId);
+        return Future.succeededFuture();
+      });
     } else {
       fileDefinitionService.save(fileExportDefinition.withStatus(FileDefinition.Status.ERROR), tenantId);
-      //update job execution with error exportresult
+      updateJobExecutionStatus(jobExecutionId, JobExecution.Status.FAIL, tenantId);
     }
   }
 
+  /**
+   * Updates jobExecution status with IN-PROGRESS && updates exported files && updates completed date
+   *
+   * @param id                   job execution id
+   * @param fileExportDefinition definition of the file to export
+   * @param tenantId             tenant id
+   */
+  private void updateJobExecutionStatusAndExportedFiles(String id, FileDefinition fileExportDefinition, String tenantId) {
+    jobExecutionService.getById(id, tenantId).compose(optionalJobExecution -> {
+      optionalJobExecution.map(jobExecution -> {
+        ExportedFile exportedFile = new ExportedFile()
+          .withFileId(UUID.randomUUID().toString())
+          .withFileName(fileExportDefinition.getFileName());
+        Set<ExportedFile> exportedFiles = jobExecution.getExportedFiles();
+        exportedFiles.add(exportedFile);
+        jobExecution.setExportedFiles(exportedFiles);
+        jobExecution.setStatus(JobExecution.Status.IN_PROGRESS);
+        jobExecution.setCompletedDate(new Date());
```

Comment:
why are we setting the completed date, when the job is still in progress?

Reply:
You are right, It is unnecessary here, will remove it